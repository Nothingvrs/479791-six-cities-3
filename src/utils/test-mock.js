export const mockTestData = {
  numberOfOffers: 32,
  cards: [
    `Canal View Prinsengracht`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`
  ]
};

export const mockCards = [
  {
    id: 0,
    name: `Wood and stone place`,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 35,
    isInBookmark: false,
    mark: 2,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,

    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 5,
    avgMark: 5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location `,
    type: `Private room`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 555,
    isInBookmark: true,
    mark: 3,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    imgs: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `img/avatar-angelina.jpg`,
      status: `pro`
    },
    addressCoords: [52.3809553943508, 4.939309666406198]
  }
];


export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};


