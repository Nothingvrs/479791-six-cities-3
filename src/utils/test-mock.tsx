import {CardModel, getCities} from './utils';
import {Authorization} from '../reducer/user/user-reducer';
import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import {createApi} from '../api';

export const userData = {
  id: 1,
  email: `laura@example.com`,
  name: `laura`,
  img: `/static/avatar/7.jpg`,
  isPro: false
};
export const mockCards: CardModel[] = [
  {
    id: 0,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 120,
    isInBookmark: true,
    mark: 4,
    isPremium: true,
    bedroomNo: 1,
    capacity: 1,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Angelina`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.85309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 1,
    name: `Wood and stone place`,
    type: `Private room`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/room.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 80,
    isInBookmark: true,
    mark: 0,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`
    ],
    descriptions: [
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.369553943508, 4.85309666406198],
    comments: [
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 2,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-02.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 132,
    isInBookmark: false,
    mark: 4,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `Chalets Alpins- 15 Chemin des Skieurs is located in Stoneham, 9.7 km from Les Marais du Nord, 9.7 km from Camping Stoneham, as well as 19.3 km from Nordique Spa Stoneham. The air-conditioned accommodation is an 8-minute walk from Halte O Spa, and guests benefit from private parking available on site and free WiFi.`,
      `Aventures Nord-Bec is 24.1 km from the chalet, while Jacques-Cartier National Park is 48.3 km from the property. The nearest airport is Qu√©bec City Jean Lesage International Airport, 40.2 km from Chalets Alpins- 15.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Olga`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3909553943508, 4.929309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  },
  {
    id: 3,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    previewImg: `/img/apartment-01.jpg`,
    imgs: [
      `/img/apartment-03.jpg`,
      `/img/apartment-01.jpg`,
      `/img/apartment-02.jpg`,
      `/img/apartment-03.jpg`,
      `/img/studio-01.jpg`,
      `/img/room.jpg`
    ],
    price: 180,
    isInBookmark: false,
    mark: 5,
    isPremium: false,
    bedroomNo: 3,
    capacity: 4,
    facilities: [
      `Wi-Fi`,
      `Heating`,
      `Kitchen`,
      `Fridge`,
      `Washing machine`,
      `Coffee machine`,
      `Dishwasher`,
      `Towels`,
      `Baby seat`,
      `Cabel TV`
    ],
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
      `The chalet features 5 separate bedrooms, 1 bathroom, a fully equipped kitchen with a dining area and dishwasher, and a living room with a TV.`
    ],
    reviews: 2,
    avgMark: 4.5,
    hostUser: {
      name: `Natalia`,
      img: `/img/avatar-angelina.jpg`,
      isPro: true
    },
    addressCoords: [52.3809553943508, 4.939309666406198],
    comments: [
      {
        id: 1,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 1,
        date: `2020-03-26T13:51:50.417Z`,
        user: userData
      },
      {
        id: 2,
        comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        mark: 4,
        date: `2020-01-26T13:51:50.417Z`,
        user: userData
      }
    ],
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    }
  }
];

export const mockCardWithServerFormat = {
  id: 3,
  title: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  preview_image: `/img/apartment-01.jpg`,
  images: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  /* eslint @typescript-eslint/camelcase: ["error", {properties: "never"}] */
  is_favorite: false,
  rating: 5,
  is_premium: false,
  bedrooms: 3,
  max_adults: 4,
  goods: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  reviews: 2,
  avgMark: 4.5,
  host: {
    id: 1,
    name: `Natalia`,
    avatar_url: `/img/avatar-angelina.jpg`,
    is_pro: true
  },
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  location: {
    latitude: 52.3809553943508,
    longitude: 4.939309666406198,
    zoom: 13
  }
};

export const mockCommentWithServerFormat = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  rating: 4,
  user: {
    avatar_url: `img/1.png`,
    id: 4,
    is_pro: false,
    name: `Max`
  }
};

export const mockCommentAdapted = {
  comment:
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`,
  id: 1,
  mark: 4,
  user: {
    img: `img/1.png`,
    email: undefined,
    id: 4,
    isPro: false,
    name: `Max`
  }
};

export const adaptedMockData = {
  id: 3,
  name: `Nice, cozy, warm big bed apartment`,
  type: `Apartment`,
  previewImg: `/img/apartment-01.jpg`,
  imgs: [
    `/img/apartment-03.jpg`,
    `/img/apartment-01.jpg`,
    `/img/apartment-02.jpg`,
    `/img/apartment-03.jpg`,
    `/img/studio-01.jpg`,
    `/img/room.jpg`
  ],
  price: 180,
  isInBookmark: false,
  mark: 5,
  isPremium: false,
  bedroomNo: 3,
  capacity: 4,
  facilities: [
    `Wi-Fi`,
    `Heating`,
    `Kitchen`,
    `Fridge`,
    `Washing machine`,
    `Coffee machine`,
    `Dishwasher`,
    `Towels`,
    `Baby seat`,
    `Cabel TV`
  ],
  descriptions: [
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  ],
  hostUser: {
    id: 1,
    name: `Natalia`,
    img: `/img/avatar-angelina.jpg`,
    isPro: true
  },
  addressCoords: [52.3809553943508, 4.939309666406198],
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  }
};

export const mockCities = [
  {
    name: `Amsterdam`,
    location: {
      latitude: 52.37454,
      longitude: 4.897976,
      zoom: 13
    }
  },
  {
    name: `Dusseldorf`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  }
];

export const logInMockData = {
  password: `123`,
  email: `1513@123.ru`
};

export const mockComment = {
  id: 1,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
  mark: 4,
  date: `2020-01-26T13:51:50.417Z`,
  user: userData
};

export const findByTestAtr = (component, attr) => {
  return component.find(`[data-test='${attr}']`);
};

export const getTestStore = () => {
  const initialState = {
    data: {
      city: mockCities[0],
      offers: mockCards,
      citiesNames: getCities(mockCards),
      hoveredId: -1,
      filterName: `popular`
    },
    user: userData,
    authorizationStatus: Authorization.NO_AUTH
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();
  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
