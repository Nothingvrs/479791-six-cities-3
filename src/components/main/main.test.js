import React from 'react';
import Main from './main';
import {mockCards} from '../../utils/test-mock';
import Enzyme, {mount} from "enzyme";
import EnzymeReactAdapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

Enzyme.configure({adapter: new EnzymeReactAdapter()});
it(`Main successfully rendered`, () => {
  const div = global.document.createElement(`div`);
  global.document.body.appendChild(div);
  const tree = mount(<Main cards ={mockCards} onCardHover = {() => {}} onHeaderClick = {() => {}} />, {attachTo: div});
  expect(toJson(tree, {mode: `deep`})).toMatchSnapshot();
});
