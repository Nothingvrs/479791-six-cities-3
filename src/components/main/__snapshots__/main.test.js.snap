// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main successfully rendered 1`] = `
<Connect(Main)
  cards={
    Array [
      Object {
        "addressCoords": Array [
          52.3909553943508,
          4.85309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 1,
        "capacity": 1,
        "city": "Amsterdam",
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 1,
            "mark": 4,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "name": "Angelina",
          "status": "pro",
        },
        "id": 0,
        "imgs": Array [
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": true,
        "isPremium": true,
        "mark": 4,
        "name": "Beautiful & luxurious apartment at great location",
        "nearOffers": Array [
          1,
          2,
          3,
        ],
        "price": 120,
        "reviews": 2,
        "type": "Apartment",
      },
      Object {
        "addressCoords": Array [
          52.369553943508,
          4.85309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": "Hamburg",
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "name": "Natalia",
          "status": "pro",
        },
        "id": 1,
        "imgs": Array [
          "/img/room.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": true,
        "isPremium": false,
        "mark": 0,
        "name": "Wood and stone place",
        "nearOffers": Array [
          0,
          2,
          3,
        ],
        "price": 80,
        "reviews": 2,
        "type": "Private room",
      },
      Object {
        "addressCoords": Array [
          52.3909553943508,
          4.929309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": "Hamburg",
        "comments": Array [
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 1,
            "mark": 4,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "Chalets Alpins- 15 Chemin des Skieurs is located in Stoneham, 9.7 km from Les Marais du Nord, 9.7 km from Camping Stoneham, as well as 19.3 km from Nordique Spa Stoneham. The air-conditioned accommodation is an 8-minute walk from Halte O Spa, and guests benefit from private parking available on site and free WiFi.",
          "Aventures Nord-Bec is 24.1 km from the chalet, while Jacques-Cartier National Park is 48.3 km from the property. The nearest airport is QuÃ©bec City Jean Lesage International Airport, 40.2 km from Chalets Alpins- 15.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
          "Cabel TV",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "name": "Olga",
          "status": "pro",
        },
        "id": 2,
        "imgs": Array [
          "/img/apartment-02.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": false,
        "isPremium": false,
        "mark": 4,
        "name": "Canal View Prinsengracht",
        "nearOffers": Array [
          0,
          1,
          3,
        ],
        "price": 132,
        "reviews": 2,
        "type": "Apartment",
      },
      Object {
        "addressCoords": Array [
          52.3809553943508,
          4.939309666406198,
        ],
        "avgMark": 4.5,
        "bedroomNo": 3,
        "capacity": 4,
        "city": "Amsterdam",
        "comments": Array [
          Object {
            "author": "Alex",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-03-26T13:51:50.417Z",
            "id": 1,
            "mark": 1,
          },
          Object {
            "author": "Max",
            "authorImg": "/img/avatar-max.jpg",
            "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
            "date": "2020-01-26T13:51:50.417Z",
            "id": 2,
            "mark": 4,
          },
        ],
        "descriptions": Array [
          "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.",
          "An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
          "The chalet features 5 separate bedrooms, 1 bathroom, a fully equipped kitchen with a dining area and dishwasher, and a living room with a TV.",
        ],
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
          "Fridge",
          "Washing machine",
          "Coffee machine",
          "Dishwasher",
          "Towels",
          "Baby seat",
          "Cabel TV",
        ],
        "hostUser": Object {
          "img": "/img/avatar-angelina.jpg",
          "name": "Natalia",
          "status": "standard",
        },
        "id": 3,
        "imgs": Array [
          "/img/apartment-03.jpg",
          "/img/apartment-01.jpg",
          "/img/apartment-02.jpg",
          "/img/apartment-03.jpg",
          "/img/studio-01.jpg",
          "/img/room.jpg",
        ],
        "isInBookmark": false,
        "isPremium": false,
        "mark": 5,
        "name": "Nice, cozy, warm big bed apartment",
        "nearOffers": Array [
          0,
          1,
          2,
        ],
        "price": 180,
        "reviews": 2,
        "type": "Apartment",
      },
    ]
  }
  location={
    Object {
      "location": Object {
        "pathname": "",
        "search": "",
      },
    }
  }
  onHeaderClick={[Function]}
>
  <div
    className="page page--gray page--main"
  >
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="header__wrapper"
        >
          <div
            className="header__left"
          >
            <a
              className="header__logo-link header__logo-link--active"
            >
              <img
                alt="6 cities logo"
                className="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            className="header__nav"
          >
            <ul
              className="header__nav-list"
            >
              <li
                className="header__nav-item user"
              >
                <a
                  className="header__nav-link header__nav-link--profile"
                  href="#"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    className="header__user-name user__name"
                  >
                    Oliver.conner@gmail.com
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      className="page__main page__main--index"
    >
      <h1
        className="visually-hidden"
      >
        Cities
      </h1>
      <div
        className="tabs"
      >
        <section
          className="locations container"
        >
          <ul
            className="locations__list tabs__list"
          >
            <li
              className="locations__item"
              key="Amsterdam - 0"
            >
              <a
                className="locations__item-link tabs__item"
                data-test="test-city-click"
                href="Amsterdam"
                onClick={[Function]}
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              className="locations__item"
              key="Hamburg - 1"
            >
              <a
                className="locations__item-link tabs__item"
                data-test="test-city-click"
                href="Hamburg"
                onClick={[Function]}
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        className="cities"
      >
        <div
          className="cities__places-container container"
        >
          <section
            className="cities__places places"
          >
            <h2
              className="visually-hidden"
            >
              Places
            </h2>
            <b
              className="places__found"
            >
              2
               places to stay in 
              Amsterdam
            </b>
            <form
              action="#"
              className="places__sorting"
              method="get"
            >
              <span
                className="places__sorting-caption"
              >
                Sort by
              </span>
              <span
                className="places__sorting-type"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
                <svg
                  className="places__sorting-arrow"
                  height="4"
                  width="7"
                >
                  <use
                    xlinkHref="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                className="places__options places__options--custom places__options--custom"
              >
                <li
                  className="places__option places__option--active"
                  data-test="test-filter-click"
                  key="popular - 0"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Popular
                </li>
                <li
                  className="places__option false"
                  data-test="test-filter-click"
                  key="lowToHigh - 1"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Price: low to high
                </li>
                <li
                  className="places__option false"
                  data-test="test-filter-click"
                  key="highToLow - 2"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Price: high to low
                </li>
                <li
                  className="places__option false"
                  data-test="test-filter-click"
                  key="topRated - 3"
                  onClick={[Function]}
                  tabIndex="0"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              className="cities__places-list places__list tabs__content"
            >
              Array [
                <article
                  className="cities__place-card place-card"
                  data-test="test-card-hover"
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                >
                  <div
                    className="place-card__mark"
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    className="cities__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="200"
                        src="/img/apartment-01.jpg"
                        width="260"
                      />
                    </a>
                  </div>
                  <div
                    className="place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          â¬
                          120
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          /Â night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button button place-card__bookmark-button--active"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          In
                           bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "90%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    >
                      <a
                        data-test="test-header-click"
                        href="#"
                        onClick={[Function]}
                      >
                        Beautiful & luxurious apartment at great location
                      </a>
                    </h2>
                    <p
                      className="place-card__type"
                    >
                      Apartment
                    </p>
                  </div>
                </article>,
                <article
                  className="cities__place-card place-card"
                  data-test="test-card-hover"
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                >
                  <div
                    className="cities__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="200"
                        src="/img/apartment-03.jpg"
                        width="260"
                      />
                    </a>
                  </div>
                  <div
                    className="place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          â¬
                          180
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          /Â night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button button false"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          To
                           bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "90%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    >
                      <a
                        data-test="test-header-click"
                        href="#"
                        onClick={[Function]}
                      >
                        Nice, cozy, warm big bed apartment
                      </a>
                    </h2>
                    <p
                      className="place-card__type"
                    >
                      Apartment
                    </p>
                  </div>
                </article>,
              ]
            </div>
          </section>
          <div
            className="cities__right-section"
          >
            <section
              className="cities__map map"
            />
          </div>
        </div>
      </div>
    </main>
  </div>
</Connect(Main)>
`;
